Lib/test/test_pdb.py

```diff
@@ -1807,10 +1811,47 @@ def test_issue42383(self):
             self.assertEqual(stdout.split('\n')[6].rstrip('\r'), expected)
 
 
+class ChecklineTests(unittest.TestCase):
+    def setUp(self):
+        linecache.clearcache()  # Pdb.checkline() uses linecache.getline()
+
+    def tearDown(self):
+        os_helper.unlink(os_helper.TESTFN)
```

Comment:
If you make ChecklineTests a subclass of PdbTestCase then you don't need the teardown here.

Reply:
That would double pull in all of the tests in `PdbTestCase` as well.

Reply:
We could make a top-level test class for both of them, but I don't know if it's worth it to save those lines.

Reply:
You're right, of course. Leave it as it is.